src/utils/units.cpp

```diff
@@ -0,0 +1,100 @@
+//========================================================================================
+// (C) (or copyright) 2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include "units.hpp"
+
+namespace ArtemisUtils {
+
+Units::Units(ParameterInput *pin) {
+  unit_system_ = pin->GetOrAddString("artemis/units", "type", "code");
+  if (unit_system_ == "code") {
+    length_ = 1.;
+    time_ = 1.;
+    mass_ = 1.;
+    G_ = 1.;
+    kb_ = 1.;
+    c_ = 1.;
+    h_ = 1.;
+    Msolar_ = 1.;
+    AU_ = 1.;
+  } else {
+    if (unit_system_ == "cgs") {
+      parthenon::constants::PhysicalConstants<parthenon::constants::CGS> pc;
```

Comment:
Should we be grabbing all the constants in parthenon's constants?

Reply:
I was just taking what we need; parthenon's constants is pretty thorough. Also because this class is runtime and be copied to device, it probably is best to keep it as small as possible.